tools/cli/src/mentat_cli/command_parser.rs

```diff
@@ -110,16 +112,21 @@ pub fn command(s: &str) -> Result<Command, cli::Error> {
                             Ok((d.to_string() + input, Consumed::Empty(input)))
                         })));
 
-    let query_parser = try(string(QUERY_COMMAND)).or(try(string(ALT_QUERY_COMMAND)))
+    let edn_arg_parser = || spaces()
```

Comment:
I don't see any movement or discussion of https://github.com/mozilla/mentat/pull/465#discussion_r118084725 -- did you try and fail to improve this, or do you particularly like the expression as is?

Improving this would avoid `.into_iter().collect()` two places below, which is a strong sign that your types are wrong: IIUC, you're collecting `Vec<String>` (where each string is one character!) into a `String`.  That's going to inefficient.